server/internal/infrastructure/gcp/cloudtasks.go

```diff
@@ -5,74 +5,64 @@ import (
 	"encoding/json"
 
 	cloudtasks "cloud.google.com/go/cloudtasks/apiv2"
+	"github.com/googleapis/gax-go/v2"
 	"github.com/reearth/reearth-cms/server/internal/usecase/gateway"
-	"github.com/reearth/reearth-cms/server/pkg/id"
-	"github.com/reearth/reearth-cms/server/pkg/id/idx"
 	"github.com/reearth/reearth-cms/server/pkg/task"
-	"google.golang.org/api/option"
 	taskspb "google.golang.org/genproto/googleapis/cloud/tasks/v2"
 )
 
 type TaskRunner struct {
 	queuePath     string
 	subscriberURL string
-	credFilePath  string
+	c             *cloudtasks.Client
 }
 
-func NewTaskRunner(c *CloudTasksConfig, opts ...TaskRunnerOption) (gateway.TaskRunner, error) {
-	opts2 := defaultTaskRunnerOptions()
-	for _, o := range opts {
-		o.Apply(opts2)
+func NewTaskRunner(ctx context.Context, c *CloudTasksConfig) (gateway.TaskRunner, error) {
+	qURL, err := c.buildQueueUrl()
+	if err != nil {
+		return nil, err
 	}
 
-	qURL, err := c.buildQueueUrl()
+	cl, err := cloudtasks.NewClient(ctx)
 	if err != nil {
 		return nil, err
 	}
 
 	return &TaskRunner{
 		queuePath:     qURL,
+		c:             cl,
 		subscriberURL: c.SubscriberURL,
-		credFilePath:  opts2.credFilePath,
 	}, nil
 }
 
 // Run implements gateway.TaskRunner
-func (t *TaskRunner) Run(ctx context.Context, p task.Payload) (id.TaskID, error) {
-
-	client, closeFn, err := t.client(ctx)
-	if err != nil {
-		return idx.ID[id.Task]{}, err
-	}
-	defer closeFn()
-
+func (t *TaskRunner) Run(ctx context.Context, p task.Payload) error {
 	bPayload, err := json.Marshal(p.DecompressAsset.Payload())
 	if err != nil {
-		return idx.ID[id.Task]{}, err
+		return err
 	}
 	req := t.buildRequest(t.subscriberURL, bPayload)
 
-	_, err = client.CreateTask(ctx, req)
+	_, err = t.CreateTask(ctx, req)
 	if err != nil {
-		return idx.ID[id.Task]{}, err
+		return err
 	}
 
-	//TODO: impl here
-	return idx.ID[id.Task]{}, nil
+	return nil
 }
 
-func (t *TaskRunner) client(ctx context.Context) (*cloudtasks.Client, func(), error) {
-	var c *cloudtasks.Client
-	var err error
-	if t.credFilePath == "" {
-		c, err = cloudtasks.NewClient(ctx)
-	} else {
-		c, err = cloudtasks.NewClient(ctx, option.WithCredentialsFile(t.credFilePath))
-	}
-	if err != nil {
-		return nil, nil, err
-	}
-	return c, func() { c.Close() }, nil
+// setClient is intended to be used for testing to inject client from external
+func (t *TaskRunner) setClient(c *cloudtasks.Client) {
+	t.c = c
+}
+
+// CloseConn is the function to close cloudtasks Client's connection. We expect this function is prepared for interactive connection since GCP SDK uses gRPC internally. To avoid instantiate client everytime, we keep the client's instance.
+func (t *TaskRunner) CloseConn() error {
+	return t.c.Close()
+}
+
+func (t *TaskRunner) CreateTask(ctx context.Context, req *taskspb.CreateTaskRequest, opts ...gax.CallOption) (*taskspb.Task, error) {
```

Comment:
```suggestion
func (t *TaskRunner) createTask(ctx context.Context, req *taskspb.CreateTaskRequest, opts ...gax.CallOption) (*taskspb.Task, error) {
```

it should be a private method